### PR TITLE
samples: suit: Documentation update

### PR DESCRIPTION
**Pull request by jnsgsr**
Correct build and test steps in the documentation.

### CODE REVIEW

**Review by ahasztag — APPROVED** [review]

**Review by e-rk — APPROVED** [review]

**Comment by shanthanordic** [comment]
who is the tech doc writer who should approve this ?

**Comment by jnsgsr** [comment]
> who is the tech doc writer who should approve this ?

unfortunately it is not clear, I will add ncs-doc-owners hopefully and b-gent someone can find time to review

**Comment by greg-fer** [comment]
> > who is the tech doc writer who should approve this ?
> 
> unfortunately it is not clear, I will add ncs-doc-owners and b-gent hopefully someone can find time to review

This is SUIT, so @kittydepa or @richabp . But also nRF54H, so @FrancescoSer needs to be tagged (as he is).

File: samples/suit/smp_transfer/README.rst [hunk 1]
```diff
@@ -118,15 +118,15 @@ You can enable the external flash support by setting the following ``FILE_SUFFIX
 
 .. code-block:: console
 
-   west build -p -b nrf54h20dk/nrf54h20/cpuapp -- -DFILE_SUFFIX="extflash"
+   west build --pristine -b nrf54h20dk/nrf54h20/cpuapp -- -DFILE_SUFFIX="extflash"
```
Comment (greg-fer) [inline samples/suit/smp_transfer/README.rst:121]: There are 80 other occurrences in the NCS where `-p` is used, but only here we're expanding it. 
What is the reason for this change?

Comment (jnsgsr) [reply samples/suit/smp_transfer/README.rst:121]: I thought that will be an improvement to make the difference more visible, in commands where we only build an updated image it is beneficial to not have a pristine build. Should I change it to `-p` for this to be consistent in NCS?

Comment (greg-fer) [reply samples/suit/smp_transfer/README.rst:121]: Yes, this is a good change, but we should probably make it for all 80+ commands. I suggest you revert it here and we open a separate PR after the release.

Comment (jnsgsr) [reply samples/suit/smp_transfer/README.rst:121]: Yeah that will be better, thanks.

**Review by tomchy — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Review by kittydepa — APPROVED** [review]

**Review by b-gent — APPROVED** [review]
